### DISCUSSION TITLE
'Framing' needed for Calc groups added to DirectLake models

### DISCUSSION THREAD

**Discussion by m-kovalsky** [TabularEditor/TabularEditor3#1134] [Ideas]
**Describe the issue**
After creating/modifying a Calculation Group in a DirectLake model, a 'framing' operation is needed. TE3 should give a warning that the Calc Group table is unprocessed and needs to be processed in order to work. 

** Which version are you currently using? **
Tabular Editor 3, version: 3.11.0
Tabular Editor 3 edition: Enterprise

1. Open a DirectLake model connected to the PBI service
2. Create a calc group & calc item(s)
3. Save the model back to the service
4. No warning shown regarding the need to process the calc group

**Desktop (please complete the following information):**
 - Windows version: 11

**Comment by otykier** [comment]
Hi Michael.

Is this any different than the normal behavior for calc groups on import models, where a calculate refresh is also required before the calc group can be used (for example when calc items have been added/removed/renamed)?

In general, we don't show any warnings when there are objects that require a refresh - but that could be a good idea for a future update. Perhaps just as an information in the **Messages** view.

**Comment by m-kovalsky** [comment]
It is the same behavior. I just wanted to call it out because I think many people think there is no need for processing tables for a model in DirectLake mode and I think many people also don't know that a Calc Group functions like an import mode table and therefore needs processing. 

I think it'd be great to show which tables/partitions need processing for models which are connected directly to a server/workspace. Messages view seems like a good place for this.

**Comment by otykier** [comment]
Thanks! I'll add this to the backlog as an enhancement :)
